tests/util.py

```diff
@@ -36,19 +40,56 @@ def get_directories(path):
 
 
 async def protoc(
-    path: Union[str, Path], output_dir: Union[str, Path], reference: bool = False
+    path: Union[str, Path],
+    output_dir: Union[str, Path],
+    reference: bool = False,
+    pydantic_dataclasses: bool = False,
 ):
     path: Path = Path(path).resolve()
     output_dir: Path = Path(output_dir).resolve()
     python_out_option: str = "python_betterproto_out" if not reference else "python_out"
-    command = [
-        sys.executable,
-        "-m",
-        "grpc.tools.protoc",
-        f"--proto_path={path.as_posix()}",
-        f"--{python_out_option}={output_dir.as_posix()}",
-        *[p.as_posix() for p in path.glob("*.proto")],
-    ]
+
+    if pydantic_dataclasses:
+        plugin_path = Path("src/betterproto/plugin/main.py")
+
+        if "Win" in platform.system():
+            with tempfile.NamedTemporaryFile(
+                "w", encoding="UTF-8", suffix=".bat", delete=False
+            ) as tf:
+                # See https://stackoverflow.com/a/42622705
+                tf.writelines(
+                    [
+                        "@echo off",
+                        f"\nchdir {os.getcwd()}",
+                        f"\npython -u {plugin_path.as_posix()}",
+                    ]
+                )
+
+                tf.flush()
+
+                plugin_path = Path(tf.name)
+                atexit.register(os.remove, plugin_path)
```

Comment:
This is rather cursed but if you don't have a better work around I guess it's fine